### PR TITLE
refactor artifact signed url protobuf endpoints

### PR DESCRIPTION
**Pull request by lilwilsond2**

### CODE REVIEW

File: protobuf/endpoints/batch_create_artifact.proto [hunk 1]
```diff
@@ -2,9 +2,17 @@ syntax = "proto3";
...
+  // TODO(hayden@preemo.io, 04/17/2023): should these be ARTFACT_TYPE... or ARTIFACTTYPE_...
+  ARTIFACT_TYPE_UNSPECIFIED = 0;
+  ARTIFACT_TYPE_PARAMS = 1;
+  ARTIFACT_TYPE_RESULT = 2;
+}
+
 // This config is included for potential future use.
 // In the future it may include part count, expiration, or other metadata.
 message CreateArtifactConfig {
+  ArtifactType type = 1;
```
Comment (lilwilsond2) [inline protobuf/endpoints/batch_create_artifact.proto:15]: i'm not sure we're ever going to have dissimilar types, maybe this should just go on the request?

File: protobuf/endpoints/batch_create_artifact.proto [hunk 2]
```diff
@@ -2,9 +2,17 @@ syntax = "proto3";
 
 package endpoints;
 
+enum ArtifactType {
+  // TODO(hayden@preemo.io, 04/17/2023): should these be ARTFACT_TYPE... or ARTIFACTTYPE_...
```
Comment (atmiguel) [inline protobuf/endpoints/batch_create_artifact.proto:6]: i believe you did it correctly

File: protobuf/endpoints/batch_create_artifact.proto (hunk 1 above)
Comment (atmiguel) [reply protobuf/endpoints/batch_create_artifact.proto:15]: For our protobuf interfaces we should prefer whichever option reduces the likelihood of having to change/update it in the future. So I think how you have it now makes perfect sense

Comment (atmiguel) [inline protobuf/endpoints/batch_create_artifact.proto:15]: we have a convention that all fields are marked optional

Comment (lilwilsond2) [reply protobuf/endpoints/batch_create_artifact.proto:15]: got bamboozeld by the maps not being optional, but i guess protobuf isn't a fan of that?

Comment (atmiguel) [reply protobuf/endpoints/batch_create_artifact.proto]: `map` and `repeated` cannot be `optional` because they can be empty

Comment (lilwilsond2) [reply protobuf/endpoints/batch_create_artifact.proto]: i guess that makes sense

File: protobuf/endpoints/batch_create_artifact.proto [hunk 3]
```diff
@@ -2,9 +2,18 @@ syntax = "proto3";
...
+  ARTIFACT_TYPE_UNSPECIFIED = 0;
+  ARTIFACT_TYPE_PARAMS = 1;
+  ARTIFACT_TYPE_RESULT = 2;
+}
+
 message CreateArtifactConfig {
+  // Required field representing the type of content the artifact will contain.
+  optional ArtifactType type = 1;
+
+  // Optional field representing the initial number of parts to allocate for this artifact.
```
Comment (atmiguel) [inline protobuf/endpoints/batch_create_artifact.proto:15]: or should this field be required?

Comment (lilwilsond2) [reply protobuf/endpoints/batch_create_artifact.proto:15]: I thought required was toxic

Comment (atmiguel) [reply protobuf/endpoints/batch_create_artifact.proto:15]: I don't mean literally. `proto3` doesn't even have `required`. I mean semantically, like according to our documentation

Comment (lilwilsond2) [reply protobuf/endpoints/batch_create_artifact.proto:15]: Oh. I don't think we need to

**Review by atmiguel — APPROVED** [review]

**Pull request merged**